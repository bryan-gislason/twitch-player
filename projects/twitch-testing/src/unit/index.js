import { ThemeProvider } from "@twitch-player/themes/dist/context";
import { light as lightTheme } from "@twitch-player/themes/dist/themes";
import { OverridesProvider } from "@twitch-player/ui/dist/context";
import React from "react";
import { act } from "react-dom/test-utils";
import { MemoryRouter as Router } from "react-router";
import { fireEvent, render } from "react-testing-library";

const customRender = (node, ...options) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <OverridesProvider>
        <Router>{node}</Router>
      </OverridesProvider>
    </ThemeProvider>,
    ...options
  );
};

fireEvent["press"] = (node, init) => {
  fireEvent.mouseDown(node, init);
  fireEvent.mouseUp(node, init);
};

fireEvent["resize"] = (node, init) => {
  const { target: { left = 0, top = 0, width = 0, height = 0 } = {} } = init;

  Object.defineProperties(window.HTMLElement.prototype, {
    offsetLeft: { get: () => left },
    offsetTop: { get: () => top },
    offsetWidth: { get: () => width },
    offsetHeight: { get: () => height },
  });

  const eventInit = { bubbles: false, cancelable: false };
  const event = new Event(node, eventInit);

  act(() => {
    fireEvent(node, event);
    // eslint-disable-next-line no-undef
    jest.runAllTimers();
  });
};

export * from "react-testing-library";

export { customRender as render };
