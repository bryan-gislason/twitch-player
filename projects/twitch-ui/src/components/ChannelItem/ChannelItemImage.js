import PropTypes from "prop-types";
import React, { memo } from "react";
import { ImageBackground } from "react-native";
import { useTheme } from "../../hooks";
import { processImageTemplate } from "../../utils";
import ChannelItemImageOverlay from "./ChannelItemImageOverlay";

const ChannelItemImage = ({
  url,
  title,
  viewersCount,
  width: previewImageWidth,
  height: previewImageHeight,
  testID = "channel-item-image",
}) => {
  const { colors, layout } = useTheme();

  const imageURL = processImageTemplate(
    url,
    layout.maxChannelWidth,
    layout.maxChannelHeight
  );

  const imageStyle = {
    backgroundColor: colors.itemBackground,
    height: previewImageHeight,
    width: previewImageWidth,
  };

  return (
    <ImageBackground
      source={{ uri: imageURL }}
      accessibilityLabel={title}
      style={imageStyle}
      testID={testID}
    >
      <ChannelItemImageOverlay
        viewersCount={viewersCount}
        testID={`${testID}-overlay`}
      />
    </ImageBackground>
  );
};

ChannelItemImage.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  viewersCount: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  testID: PropTypes.string,
};

export default memo(ChannelItemImage);
