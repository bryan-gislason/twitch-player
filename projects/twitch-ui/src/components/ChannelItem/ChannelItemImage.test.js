import { render } from "@twitch-player/testing/dist/unit";
import React from "react";
import ChannelItemImage from "./ChannelItemImage";

describe("ChannelItemImage", () => {
  let instance;

  beforeEach(() => {
    instance = render(
      <ChannelItemImage
        url="channels/live_user_ninja-800x450.jpg"
        title="We Gaming <3 "
        viewersCount={123456}
        width={400}
        height={225}
      />
    );
  });

  it("renders a channel item image", () => {
    expect(instance.getByTestId("channel-item-image")).toMatchSnapshot();
  });
});
