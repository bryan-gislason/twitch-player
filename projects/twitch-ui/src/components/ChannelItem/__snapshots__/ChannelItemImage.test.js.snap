// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelItemImage renders a channel item image 1`] = `
<div
  aria-label="We Gaming <3 "
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-overflowX-11yh6sk rn-overflowY-buy8e9 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-zIndex-1wyyakw"
  data-testid="channel-item-image"
  style="height: 225px; width: 400px;"
>
  <div
    class="rn-alignItems-1oszu61 rn-backgroundColor-1niwhzg rn-backgroundPosition-vvn4in rn-backgroundRepeat-u6sd8q rn-backgroundSize-4gszlv rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-bottom-1p0dtai rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1pi2tsx rn-left-1d2f490 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    style="background-image: url(channels/live_user_ninja-800x450.jpg);"
  />
  <img
    alt="We Gaming <3 "
    class="rn-bottom-1p0dtai rn-height-1pi2tsx rn-left-1d2f490 rn-opacity-1272l3b rn-position-u8s1d rn-right-zchlnj rn-top-ipm5af rn-width-13qz1uu rn-zIndex-1wyyakw"
    draggable="false"
    src="channels/live_user_ninja-800x450.jpg"
  />
</div>
`;
